indicator_stochastic_oscillator_test.go

```diff
@@ -0,0 +1,85 @@
+package techan
+
+import (
+	"math"
+	"testing"
+
+	"github.com/sdcoffey/big"
+	"github.com/stretchr/testify/assert"
+)
+
+var fastStochValues = []float64{
+	100,
+	100,
+	100.0 * 12.0 / 16.0,
+	100.0 * 2.0 / 16.0,
+	100.0 * 6.0 / 16.0,
+	100.0 * 2.0 / 16.0,
+	100.0 * 3.0 / 15.0,
+	100.0 * 2.0 / 16.0,
+	100.0 * 4.0 / 13.0,
+	100.0 * 11.0 / 17.0,
+	100.0 * 24.0 / 49.0,
+}
+
+func TestFastStochasticIndicator(t *testing.T) {
+	ts := mockTimeSeriesOCHL(
+		[]float64{10, 12, 12, 8},
+		[]float64{11, 14, 14, 9},
+		[]float64{10, 20, 24, 10},
+		[]float64{9, 10, 11, 9},
+		[]float64{11, 14, 14, 9},
+		[]float64{9, 10, 11, 9},
+		[]float64{10, 12, 12, 10},
+		[]float64{9, 10, 11, 8},
+		[]float64{6, 5, 8, 1},
+		[]float64{15, 12, 18, 9},
+		[]float64{35, 25, 50, 20},
+	)
+
+	window := 6
+
+	k := NewFastStochasticIndicator(ts, window)
+
+	decimalEquals(t, fastStochValues[0], k.Calculate(0))
```

Comment:
In the future, you can make these tests a little less repetitive by using the `indicatorEquals` test fn:

example:
https://github.com/sdcoffey/techan/blob/a2f4ef854b8a460ee1e084958ebad76183b87c2e/indicator_modified_moving_average_test.go#L7-L25